com.microsoft.mrtk.input/Tracking/UnboundedTrackingMode.cs

```diff
@@ -0,0 +1,178 @@
+using Microsoft.MixedReality.OpenXR.Remoting;
+#endif
+
+#if UNITYXR_MANAGEMENT_PRESENT
+using UnityEngine.XR.Management;
+#endif
+
+namespace Microsoft.MixedReality.Toolkit.Input
+{
+    /// <summary>
+    /// This component works along side Unity's <see cref="XROrigin"/> to ensure that the <see cref="XRInputSubsystem"/> has
+    /// the <see cref="TrackingOriginModeFlags.Unbounded"/> flag set on its tracking origin mode if this flag is supported.
+    /// </summary>
+    /// <remarks>
+    /// <para>
+    /// Devices like the HoloLens 2 should use unbounded reference space. This reference space enables the viewer to move
+    /// freely through a complex environment, often many meters from where they started, while always optimizing for coordinate
+    /// system stability near the viewer. For more information about unbounded space see the
+    /// <see href="https://registry.khronos.org/OpenXR/specs/1.0/html/xrspec.html#XR_MSFT_unbounded_reference_space">OpenXR Specification</see>.
+    /// </para>
+    /// <para>
+    /// The <see cref="TrackingOriginModeFlags.Unbounded"/> flag is only applied if the <see cref="XROrigin.RequestedTrackingOriginMode"/>
+    /// is set to <see cref="XROrigin.TrackingOriginMode.NotSpecified"/>, the device supports unbounded spaces, and <see cref="XRInputSubsystem"/>
+    /// current tracking mode is set to <see cref="TrackingOriginModeFlags.Device"/>.
+    /// </para> 
+    /// </remarks>
+    [RequireComponent(typeof(XROrigin))]
+    [AddComponentMenu("MRTK/Input/Unbounded Tracking Mode")]
+    public class UnboundedTrackingMode : MonoBehaviour
+    {
+#if UNITYXR_MANAGEMENT_PRESENT
+        private XRInputSubsystem m_inputSubsystem;
+        private XROrigin.TrackingOriginMode m_requestedTrackingOriginMode = XROrigin.TrackingOriginMode.NotSpecified;
+
+        /// <summary>
+        /// A Unity event function that is called when the script component has been enabled.
+        /// </summary>
+        /// <remarks>
+        /// This will attempt to set <see cref="XRInputSubsystem"/> tracking mode to <see cref="TrackingOriginModeFlags.Unbounded"/>.
+        /// </remarks>
+        private void OnEnable()
+        {
+            XRGeneralSettings xrSettings = XRGeneralSettings.Instance;
+            if (xrSettings == null)
+            {
+                Debug.LogWarning($"EyeLevelSceneOrigin: XRGeneralSettings is null.");
+                return;
+            }
+
+            XRManagerSettings xrManager = xrSettings.Manager;
+            if (xrManager == null)
+            {
+                Debug.LogWarning($"EyeLevelSceneOrigin: XRManagerSettings is null.");
+                return;
+            }
+
+            XRLoader xrLoader = xrManager.activeLoader;
+            if (xrLoader == null)
+            {
+                Debug.LogWarning($"EyeLevelSceneOrigin: XRLoader is null.");
+                return;
+            }
+
+            m_inputSubsystem = xrLoader.GetLoadedSubsystem<XRInputSubsystem>();
```

Comment:
Short answer, that's how the MRPlugin did it ;)   (copy & paste)

Thank you for the suggestion.  I'll use XRSubsystemHelpers.InputSubsystem.  Great point!

Reply:
Makes sense!

Reply:
Just tried this out... HAR apparently connects later than the physical device.   So on HAR the sub system doesn't have input system handy by the time UnboundedTrackingMode is loaded and need to listen for input event.

While all on the other hand, XRLoader has the input system handy at start.

It looks like XROrigin worked around this by querying for the input system until it's not null.  However, since MR Plugin is already requiring `UnityEngine.XR.Management` and this method has been tested and used by the MR Plugin, I'm going to stick with XRLoader for now.   The dependency on `UnityEngine.XR.Management` is not mandatory for MRTK Input, so this usage shouldn't break other platforms.

Reply:
> While all on the other hand, XRLoader has the input system handy at start.

Oh, that's interesting! I didn't think XRManagerSettings would have an `activeLoader` or an XRLoader would have any loaded subsystems until remoting connects and the loader is initialized, but I must be mistaken.